js/misc/missionGameService.js

```diff
@@ -29,7 +29,7 @@ function callWhenComplete(completion, key, callback) {
     });
 
     if (complete && callback) {
-        callback(key);
+        callback(key. arguments);
```

Comment:
Actually, this was a typo -> `.` should be `,` . Nonetheless, I'm not sure if `arguments` works that way now that I think of it.  I think you need to do `callback.apply(this, [key].concat(Array.from(arguments)));`